include/powsybl/iidm/NetworkVariant.hpp

```diff
@@ -9,23 +9,28 @@
 #define POWSYBL_IIDM_NETWORKVARIANT_HPP
 
 #include <powsybl/iidm/Variant.hpp>
+#include <powsybl/iidm/VariantArray.hpp>
 
 namespace powsybl {
 
 namespace iidm {
 
+class Network;
+
 namespace network {
 
-class VariantImpl : public Variant<VariantImpl> {
+class VariantImpl : public Variant<Network, VariantImpl> {
 public: // Variant
     std::unique_ptr<VariantImpl> copy() const override;
 
 public:
-    VariantImpl() = default;
+    VariantImpl(Network& network);
 
     VariantImpl(const VariantImpl& variant) = delete;
 
-    VariantImpl(VariantImpl&&) noexcept = default;
+    VariantImpl(VariantImpl&& variant) noexcept = delete;
```

Comment:
To be discussed : `variant` parameter name is useless, we did not write it usually